redis/pool_test.go

```diff
@@ -368,21 +396,92 @@ func TestPoolConcurrenSendReceive(t *testing.T) {
 }
 
 func TestPoolBorrowCheck(t *testing.T) {
-	d := poolDialer{t: t}
-	p := &redis.Pool{
-		MaxIdle:      2,
-		Dial:         d.dial,
-		TestOnBorrow: func(redis.Conn, time.Time) error { return redis.Error("BLAH") },
+	pingN := func(ctx context.Context, p *redis.Pool, n int) {
+		for i := 0; i < n; i++ {
+			c, err := p.GetContext(ctx)
+			require.NoError(t, err)
+			_, err = redis.DoContext(c, ctx, "PING")
+			require.NoError(t, err)
+			require.NoError(t, c.Close())
```

Comment:
defer the close so it will always be executed even if one of the other tests fail.
```suggestion
			c, err := p.GetContext(ctx)
			require.NoError(t, err)
			defer func() {
				require.NoError(t, c.Close())
			}()

			_, err = redis.DoContext(c, ctx, "PING")
			require.NoError(t, err)
```

Reply:
Ok, done.